cql3/statements/role-management-statements.cc

```diff
@@ -118,6 +118,29 @@ future<> create_role_statement::grant_permissions_to_creator(const service::clie
     });
 }
 
+static void validate_timeout_options(const auth::custom_options& role_options) {
+    if (role_options.empty()) {
+        return;
+    }
+    auto check_duration = [&] (const sstring& repr) {
+        data_value v = duration_type->deserialize(duration_type->from_string(repr));
+        cql_duration duration = static_pointer_cast<const duration_type_impl>(duration_type)->from_value(v);
+        if (duration.months || duration.days || duration.nanoseconds > 86'400'000'000'000) {
+            throw exceptions::invalid_request_exception("Timeout values cannot be longer than 24h");
+        }
+        if (duration.nanoseconds % 1'000'000 != 0) {
+            throw exceptions::invalid_request_exception("Timeout values must be expressed in millisecond granularity");
+        }
```

Comment:
Let's also check they are strictly positive.

Reply:
I'd prefer non-negative. The reason being that while having a timeout value of 0 makes little sense on production clusters, it's great for deterministic tests.